src/main/java/com/kumuluz/ee/jwt/auth/cdi/JWTContextInfo.java

```diff
@@ -68,14 +74,18 @@
     @PostConstruct
     public void init() {
         ConfigurationUtil config = ConfigurationUtil.getInstance();
-        String publicKey;
-        publicKey = config.get("mp.jwt.verify.publickey")
-                .orElse(config.get("kumuluzee.jwt-auth.public-key").orElse(null));
 
-        if (publicKey == null) {
+        Optional<String> mpPublicKey = config.get("mp.jwt.verify.publickey");
+        String publicKey = mpPublicKey.orElse(config.get("kumuluzee.jwt-auth.public-key").orElse(null));
+
+        if ((mpPublicKey.isPresent() && mpPublicKey.get().startsWith("{location=")) || publicKey == null) {
```

Comment:
Do you think we can use something more standard to our configuration framework here in the first part of condition?

Something like:

```java
config.getMapKeys("mp.jwt.verify.publickey").isPresent()
```

Reply:
Great catch, will fix this.

Reply:
I did some refactoring to JWT context initialisation and have some comments: 
- Previous implementation didn't support multiple public keys in JWKS so I improved the support
- I didn't like the manual JWKS parsing so I created KumuluzJwkProvider and used JWK parser from auth0 library
- I believe [this](https://github.com/kumuluz/kumuluzee-jwt-auth/pull/17/commits/e74819f9255d679420ccd9062c4a8c6be60f9855#diff-713e0ca9d2e085813430b588c5d5f9a92ba914923c25417cccd7770b592226b1L116) line was there by mistake or someone tried to load json from payload with removed base64 characters?
- i think we should not set leeway (or at least not set default value) [here](https://github.com/kumuluz/kumuluzee-jwt-auth/pull/17/commits/e74819f9255d679420ccd9062c4a8c6be60f9855#diff-713e0ca9d2e085813430b588c5d5f9a92ba914923c25417cccd7770b592226b1L135) since it may be provided by token claim as stated in [RFC](https://tools.ietf.org/html/rfc7519#section-4.1.5) Should we remove default value??

Please review the changes and comment.

Reply:
Hi! Something is breaking the MicroProfile TCK, I think it's related to the third point you mentioned. Could you check it out?

Regarding the last point, I don't think that the "nbf" RFC you linked is related to leeway. I think that leeway is meant as a counter for clock skew and we should leave that in, maybe reduce its default value. What do you think?

Reply:
Yes, I am looking at the tests atm.

I think that leeway value should be provided by identity provider since it has responsibility over tokens, but anyway, looks like other frameworks uses same configuration with default value of 60 seconds therefore i suggest we should set this to 1 minute.

Reply:
I agree. Let's do that then!